### DISCUSSION TITLE
SnackBar possible enhancement

### DISCUSSION THREAD

**Discussion by bijington** [xamarin/XamarinCommunityToolkit#1383] [Ideas]
I wanted to check someones thoughts whether being able to provide a custom View to the SnackBar so something similar to the sliding view at the top of the attached video could be achieved? 

The idea as mentioned would be to define a custom view for the SnackBar to render. I assume if this was to go ahead the View may be required to implement some level of interface in order to deal with lifecycle events for the general SnackBar viewing. Also I wondered whether `SnackBarOptions` could be extended or event sub-classed to allow for simple Text specific and then a more custom approach.

If this is something that could/should go in then I would be happy to write up the feature request and also put the effort in to implementing it.

https://user-images.githubusercontent.com/17139988/121436061-ccd90d00-c977-11eb-8633-e24cd1f44397.mov

**Comment by maxkoshevoi** [comment]
As of now, XCT's SnackBar is just a wrapper around native SnackBars ([iOS](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/iOS/IOSSnackBar.ios.cs#L85), [Android](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/SnackBar.android.cs#L22)) with native views, lifecycle events, styles, etc. 
What you are describing is brand new control that needs to be implemented from scratch. 

I don't think that this is in the scope of XCT's SnackBar. Did you try using popup control for it, or just some `StackLayout` with gesture recognizer (I'm not big in custom UI, but I'm sure it's easier to implement described UI yourself)?

**Reply by bijington** [reply]
@maxkoshevoi thank you for the response. Having looked around the SnackBar code I see why this wouldn't be a good fit now.

You are correct that I did use a StackLayout and then just simply animated it in and out as required. I guess the purpose of this discussion was to see if something like that could benefit XCT but I guess it might offer little gain. 

If you did eventually want something like this then I would be happy to help out with the effort

**Reply by maxkoshevoi** [reply]
My bad. Turns out, your suggestion is already possible on iOS since it doesn't have native SnackBar implementation: https://github.com/xamarin/XamarinCommunityToolkit/issues/1399#issuecomment-864616436 (but ability to provide custom view is not exposed to the user so that experience is consistent) .

As for Android, it has native SnackBar, so we use it.

**Reply by bijington** [reply]
No problem whatsoever and thanks for the response. I am happy to accept that Android limits what implementation you can provide

**Reply by bijington** [reply]
Also feel free to close this down (I can't seem to).

**Reply by maxkoshevoi** [reply]
Discussions cannot be closed, only answered =)
Thanks for your idea, it's possible that it will make it into XCT, but it needs to be a bit more defined. SnackBar also cannot be placed at the top or dismissed with a swipe to the top. It feels like a more customizable popup, maybe it worth to extend that control.

**Reply by bijington** [reply]
I agree that this isn't a good fit for SnackBar now that I have a better understanding what a SnackBar is :). I can take a look at the popup control and see if I could suggest how to enhance it. Thanks
